src/main/kotlin/com/example/delivery/service/ProductService.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.delivery.service
+
+import com.example.delivery.domain.DomainProduct
+import com.example.delivery.dto.request.CreateProductDTO
+import com.example.delivery.exception.NotFoundException
+import com.example.delivery.mapper.ProductMapper.toDomain
+import com.example.delivery.mapper.ProductMapper.toMongo
+import com.example.delivery.repository.ProductRepository
+import org.springframework.stereotype.Service
+
+@Service
+class ProductService(private val productRepository: ProductRepository) {
+
+    fun findById(id: String): DomainProduct {
+        val product = productRepository.findById(id) ?: throw NotFoundException("Product with id $id not found")
```

Comment:
`findById` - should return `null` if an entity wasn't found.
`getById` - should throw if an entity wasn't found.